apps/web/src/studio/components/workflows/preferences/StudioWorkflowSettingsSidePanel.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC, useEffect, useMemo } from 'react';
+
+import { Sidebar } from '@novu/design-system';
+import { Title } from '@novu/novui';
+import { css } from '@novu/novui/css';
+import { useFormContext } from 'react-hook-form';
+import { useParams } from 'react-router-dom';
+import { useStudioWorkflowChannelPreferences } from '../../../../hooks/workflowChannelPreferences/useStudioWorkflowChannelPreferences';
+import { useDiscover } from '../../../hooks/useBridgeAPI';
+import { WorkflowDetailFormContext } from './WorkflowDetailFormContextProvider';
+import { WorkflowSettingsSidePanelContent } from './WorkflowSettingsSidePanelContent';
+
+type StudioWorkflowSettingsSidePanelProps = { onClose: () => void };
+
+export const StudioWorkflowSettingsSidePanel: FC<StudioWorkflowSettingsSidePanelProps> = ({ onClose }) => {
+  const { data } = useDiscover();
+  const { templateId: workflowId = '' } = useParams<{ templateId: string }>();
+
+  const { isLoading, workflowChannelPreferences } = useStudioWorkflowChannelPreferences(workflowId);
+  const { setValue } = useFormContext<WorkflowDetailFormContext>();
+
+  const workflow = useMemo(() => {
+    return data?.workflows?.find((workflow) => workflow.workflowId === workflowId);
+  }, [data, workflowId]);
+
+  useEffect(() => {
+    if (workflow) {
+      setValue('general.workflowId', workflow.workflowId);
+      // TODO: how are we handling workflow name for a workflow from framework?
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** Unsure if we should just the `name` input for Studio since I don't believe we have this data for Framework at the moment‏

Reply:
Maybe the name input should be hidden for now? As we are showing it as empty + disabled